packages/firestore/src/api/credentials.ts

```diff
@@ -217,18 +215,17 @@ export class FirebaseCredentialsProvider implements CredentialsProvider {
     this.tokenListener = () => {
       this.tokenCounter++;
       this.currentUser = this.getUser();
-      this.receivedInitialUser.resolve();
-      if (this.invokeChangeListener) {
+      this.authDeferred.resolve();
+      if (this.changeListener) {
         this.asyncQueue!.enqueueRetryable(() =>
-          this.changeListener(this.currentUser)
+          this.changeListener!(this.currentUser)
```

Comment:
`changeListener` is pushed to the `AsyncQueue` when it's registered. So for the first user, it seems it will be called twice. Is it a problem? Perhaps we still need the `invokeChangeListener` flag.

Reply:
I don't think this should happen: `setCredentialListener` sets the `changeListener` but only once the `authDeferred` promise resolves. In the initial token callback the `changeListener` will be undefined